client/repo.c

```diff
@@ -589,42 +590,49 @@ TDNFDownloadUsingMetalinkResources(
         dwError = ERROR_TDNF_INVALID_PARAMETER;
         BAIL_ON_TDNF_ERROR(dwError);
     }
-    urls = ml_file->urls;
+    urlList = ml_ctx->urls;
 
-    while (urls)
+    while(urlList)
     {
-        dwError = TDNFStringEndsWith(urls->url, TDNF_REPO_METADATA_FILE_PATH);
+        urlInfo = urlList->data;
+        if(urlInfo == NULL)
+        {
+	    dwError = ERROR_TDNF_INVALID_REPO_FILE;
```

Comment:
still indentation is not right in the if {} block in this file.

Reply:
done